js/graphql.md

```diff
@@ -0,0 +1,1374 @@
+mutation CreateComment($postId:ID!) {
+  createComment(input:{content:"A comment!", commentPostId:$postId}) {
+    id
+    content
+    post {
+      id
+      title
+      blog {
+        id
+        name
+      }
+    }
+  }
+}
+
+# Get a blog, its posts, and its posts comments.
+query GetBlog($blogId:ID!) {
+  getBlog(id:$blogId) {
+    id
+    name
+    posts(filter: {
+      title: {
+        eq: "My Post!"
+      }
+    }) {
+      items {
+        id
+        title
+        comments {
+          items {
+            id
+            content
+          }
+        }
+      }
+    }
+  }
+}
+
+# List all blogs, their posts, and their posts comments.
+query ListBlogs {
+  listBlogs { # Try adding: listBlog(filter: { name: { eq: "My New Blog!" } })
+    items {
+      id
+      name
+      posts { # or try adding: posts(filter: { title: { eq: "My Post!" } })
+        items {
+          id
+          title
+          comments { # and so on ...
+            items {
+              id
+              content
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+If you want to update your API, open your project's `backend/api/~apiname~/schema.graphql` file (NOT the one in the `backend/api/~apiname~/build` folder) and edit it in your favorite code editor. You can compile the `backend/api/~apiname~/schema.graphql` by running:
+
+```
+amplify api gql-compile
+```
+
+and view the compiled schema output in `backend/api/~apiname~/build/schema.graphql`.
+
+You can then push updated changes with:
+
+```
+amplify push
+```
+
+## Directives
+
+### @model
+
+Object types that are annotated with `@model` are top-level entities in the
+generated API. Objects annotated with `@model` are stored in DynamoDB and are
+capable of being protected via `@auth`, related to other objects via `@connection`,
+and streamed into Elasticsearch via `@searchable`.
+
+#### Definition
+
+```
+directive @model(
+    queries: ModelQueryMap, 
+    mutations: ModelMutationMap
+) on OBJECT
+input ModelMutationMap { create: String, update: String, delete: String }
+input ModelQueryMap { get: String, list: String }
+```
+
+#### Usage
+
+Define a GraphQL object type and annotate it with the `@model` directive to store
+objects of that type in DynamoDB and automatically configure CRUDL queries and
+mutations.
+
+```
+type Post @model {
+    id: ID! # id: ID! is a required attribute.
+    title: String!
+    tags: [String!]!
+}
+```
+
+You may also override the names of any generated queries and mutations, or remove operations entirely.
+
+```
+type Post @model(queries: { get: "post" }, mutations: null) {
+    id: ID!
+    title: String!
+    tags: [String!]!
+}
+```
+
+This would create and configure a single query field `post(id: ID!): Post` and
+no mutation fields.
+
+#### Generates
+
+A single `@model` directive configures the following AWS resources:
+
+- An Amazon DynamoDB table with 5 r/w units. Support for auto-scaling and encryption at rest coming soon.
```

Comment:
Do not put "Support for X is coming soon" on public docs or forums, as priorities and velocity can change for delivery on features.